esp-wifi/src/wifi/mod.rs

```diff
@@ -665,6 +717,13 @@ impl WifiController {
         Ok(mode == wifi_mode_t_WIFI_MODE_STA)
     }
 
+    fn is_ap_enabled(&self) -> Result<bool, WifiError> {
+        let mut mode: esp_wifi_sys::include::wifi_mode_t = 0;
+        esp_wifi_result!(unsafe { esp_wifi_sys::include::esp_wifi_get_mode(&mut mode) })?;
+
+        Ok(mode == wifi_mode_t_WIFI_MODE_AP) // should also match APSTA which is not supported yet
```

Comment:
We can probably still add the check here already, as it will just be logically or'd with just AP mode.